uamqp/client.py

```diff
@@ -188,6 +194,13 @@ def open(self):
             outgoing_window=self._outgoing_window
         )
         self._session.begin()
+        if isinstance(self._auth, _CBSAuth):
```

Comment:
If we add that string discriminator to the auth types, we wont have to do `isinstance` checks :)